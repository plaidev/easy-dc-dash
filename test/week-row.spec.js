import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import WeekRow from '@/week-row.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('week-row', () => {
  Store.registerData([
    { date: new Date('2017-05-18 10:00'), d1: 'a', d2: true, v1: 1, v2: 10 },
    { date: new Date('2017-05-19 10:00'), d1: 'a', d2: false, v1: 2, v2: 20 },
    { date: new Date('2017-05-20 10:00'), d1: 'b', d2: false, v1: 3, v2: 30 },
    { date: new Date('2017-05-21 10:00'), d1: 'b', d2: true, v1: 5, v2: 50 }
  ]);

  const wrapper = mount(localVue.extend(WeekRow), {
    localVue,
    propsData: {
      id: 'test-week-row',
      dimension: 'date',
      reduce: 'v'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
