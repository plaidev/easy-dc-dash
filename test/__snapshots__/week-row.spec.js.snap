// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`week-row to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-week-row" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g transform="translate(40,20)">
                            <g class="axis" transform="translate(0, 150)">
                                <path class="domain"></path>
                            </g>
                            <g class="row _0" transform="translate(0,5)">
                                <rect width="0" height="31.25" fill="#bd3122" class=""></rect>
                                <text x="10" y="15.625" dy="0.35em" class="row _0">Sun</text>
                            </g>
                            <g class="row _1" transform="translate(0,41.25)">
                                <rect width="0" height="31.25" fill="#2AAB9F" class=""></rect>
                                <text x="10" y="15.625" dy="0.35em" class="row _1">Thu</text>
                            </g>
                            <g class="row _2" transform="translate(0,77.5)">
                                <rect width="0" height="31.25" fill="#54BCB2" class=""></rect>
                                <text x="10" y="15.625" dy="0.35em" class="row _2">Fri</text>
                            </g>
                            <g class="row _3" transform="translate(0,113.75)">
                                <rect width="0" height="31.25" fill="#70C7BF" class=""></rect>
                                <text x="10" y="15.625" dy="0.35em" class="row _3">Sat</text>
                            </g>
                        </g>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
